### PR TITLE
Adds an overlay to chem dispensers when inserted with a beaker

### LINKED ISSUE #18766: Chemdispenser lacks sprite for when beaker is inserted [request]
Would be nice so you can tell at a glance if it's loaded

### PR DESCRIPTION
**Pull request by beesil**
[qol] Finding your large beakers has never been easier! The beaker is added under a random nozzle, and has a different sprite depending on whether it's a glass beaker or a bluespace/stasis beaker. 

Sprites courtesy of @RealestEstate, closes #18766.

image

:cl:
- rscadd: Chem dispensers now have a visual indicator when loaded with a beaker.

### CODE REVIEW

**Comment by Irradiation** [comment]
Being able to hold somebody's head in the chem dispenser to forcefeed them everything when?

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 1]
```diff
@@ -316,6 +321,21 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
...
+
+	overlays.len = 0
+	var/image/overlay
+
+	if(istype(container, /obj/item/weapon/reagent_containers/glass/beaker/bluespace) || istype(container, /obj/item/weapon/reagent_containers/glass/beaker/noreact))
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay2")
+	else
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay1")
+
+	if(beaker)
```
Comment (Exxion) [inline code/modules/reagents/machinery/chem_dispenser.dm]: I don't get the point of this var. Couldn't you just check if `container` is null?

Comment (beesil) [reply code/modules/reagents/machinery/chem_dispenser.dm]: fixed, thanks

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -316,6 +318,21 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
...
+
+	overlays.len = 0
+	var/image/overlay
+
+	if(istype(container, /obj/item/weapon/reagent_containers/glass/beaker/bluespace) || istype(container, /obj/item/weapon/reagent_containers/glass/beaker/noreact))
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay2")
+	else
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay1")
+
+	if(container)
```
Comment (Exxion) [inline code/modules/reagents/machinery/chem_dispenser.dm:326]: Sorry it's a bit of a nitpick but you should put everything except the `overlays.len = 0` under this check
No point in generating the image if it's not going to be used

Comment (beesil) [reply code/modules/reagents/machinery/chem_dispenser.dm:326]: 👍

**Review by Exxion — APPROVED** [review]

**Pull request merged**